File: fixtures/scale/elixir/large/lib/ciphers/des64/impl.ex
defmodule Ciphers.Des64.Impl do
  alias Ciphers.Des64.Model

  def encrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 + key + Model.rounds() + Model.key_bits(), 256))
  end

  def decrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 - key - Model.rounds() - Model.key_bits() + 512, 256))
  end
end

File: fixtures/scale/haskell/large/Ciphers/Rc6128/Impl.hs
module Ciphers.Rc6128.Impl (encrypt, decrypt) where

import Ciphers.Rc6128.Model (keyBits, rounds)

encrypt :: Int -> [Int] -> [Int]
encrypt key = map (\b -> mod (b + key + rounds + keyBits) 256)

decrypt :: Int -> [Int] -> [Int]
decrypt key = map (\b -> mod (b - key - rounds - keyBits + 512) 256)
